src/test/java/edu/unc/lib/boxc/migration/cdm/PermissionsCommandIT.java

```diff
@@ -0,0 +1,119 @@
+                "-w", project.getProjectPath().toString(),
+                "permissions", "generate",
+                "-id", "default",
+                "--everyone", "canViewMetadata",
+                "--authenticated", "canViewMetadata"};
+        executeExpectSuccess(args);
+        assertDefaultMapping("default", "canViewMetadata",
+                "canViewMetadata");
+    }
+
+    @Test
+    public void generateDefaultPermissionsNone() throws Exception {
+        String[] args = new String[] {
+                "-w", project.getProjectPath().toString(),
+                "permissions", "generate",
+                "-id", "default"};
+        executeExpectSuccess(args);
+        assertDefaultMapping("default", "canViewOriginals",
+                "canViewOriginals");
+    }
+
+    @Test
+    public void generateDefaultPermissionsStaffOnly() throws Exception {
+        String[] args = new String[] {
+                "-w", project.getProjectPath().toString(),
+                "permissions", "generate",
+                "-id", "default",
+                "-so"};
+        executeExpectSuccess(args);
+        assertDefaultMapping("default", "none", "none");
+    }
+
+    @Test
+    public void generateDefaultPermissionsInvalid() throws Exception {
+        String[] args = new String[] {
+                "-w", project.getProjectPath().toString(),
+                "permissions", "generate",
+                "-id", "default",
+                "--everyone", "canViewMetadata",
+                "--authenticated", "canManage"};
+        executeExpectFailure(args);
+        assertOutputContains("Assigned role value is invalid. Must be one of the following patron roles: " +
+                "[none, canDiscover, canViewMetadata, canViewAccessCopies, canViewOriginals]");
+    }
+
+    @Test
+    public void generateDefaultPermissionsWithoutForceFlag() throws Exception {
+        FileUtils.write(project.getPermissionsPath().toFile(),
+                "default,canViewMetadata,canViewMetadata", StandardCharsets.UTF_8, true);
+
+        String[] args = new String[] {
+                "-w", project.getProjectPath().toString(),
+                "permissions", "generate",
+                "-id", "default",
+                "--everyone", "canViewMetadata",
+                "--authenticated", "canViewMetadata"};
+        executeExpectFailure(args);
+        assertOutputContains("Cannot create permissions, a file already exists. " +
+                "Use the force flag to overwrite.");
+    }
+
+    @Test
+    public void generateDefaultPermissionsWithForceFlag() throws Exception {
+        FileUtils.write(project.getPermissionsPath().toFile(),
+                "default,canViewMetadata,canViewMetadata", StandardCharsets.UTF_8, true);
+
+        String[] args = new String[] {
+                "-w", project.getProjectPath().toString(),
+                "permissions", "generate",
+                "-id", "default",
+                "--everyone", "canViewMetadata",
```

Comment:
just to demonstrate that it actually updates the file, I'd suggest setting "--everyone" to something other than "canViewMetadata" here and then add an assertion to demonstrate that the file contains the new entry rather than the old one.